odoo/addons/base/res/res_partner.py

```diff
@@ -453,16 +453,15 @@ def _fields_sync(self, values):
                 self.update_address(onchange_vals)
 
         # 2. To DOWNSTREAM: sync children
-        if self.child_ids:
+        if self.search_count([('parent_id', '=', self.id)]):
```

Comment:
For boolean results,
Is not faster self.search(..., limit=1)?
Since that doesn’t need read all records just using the first result will be return early the query.
Is not?

Reply:
search with limit=1 have sometimes side effect, because PostgreSQL is not always using the same execution plan when no limit or limit=1.

https://github.com/odoo/odoo/pull/29278#issue-236168445 as an illustration to this.

Reply:
Then exists is the better option for you check the plan execution:
https://blog.jooq.org/2016/09/14/avoid-using-count-in-sql-when-you-could-use-exists/

Reply:
@rco-odoo 
Should we add suppots for “exists” queries from the api?

Reply:
What about using the same search_count(..., limit=1)
If limit==1 then we can use exists in the query since that the limit doesn’t matter from aggregate method count(*)